doc/examples/kinematics_cost_function/src/kinematics_cost_function_tutorial.cpp

```diff
@@ -0,0 +1,256 @@
+  copy_state->setFromIK(joint_model_group, target_pose, 0.0 /* timeout */, callback_fn, opts);
+
+  std::vector<double> solution;
+  current_state->copyJointGroupPositions(joint_model_group, solution);
+
+  std::vector<double> solution_no_cost_fn;
+  copy_state->copyJointGroupPositions(joint_model_group, solution_no_cost_fn);
+
+  // Now we can use our helper function from earlier to evaluate the effectiveness of the cost function.
+  double l2_solution = compute_l2_norm(solution, start_joint_positions);
+  RCLCPP_INFO_STREAM(LOGGER, "L2 norm of the solution WITH a cost function is " << l2_solution);
+  l2_solution = compute_l2_norm(solution_no_cost_fn, start_joint_positions);
+  RCLCPP_INFO_STREAM(LOGGER, "L2 norm of the solution WITHOUT a cost function is " << l2_solution);
+
+  // If we're happy with the solution, we can set it as a joint value target.
+  move_group.setJointValueTarget(solution);
+
+  // We lower the allowed maximum velocity and acceleration to 5% of their maximum.
+  // The default values are 10% (0.1).
+  // Set your preferred defaults in the joint_limits.yaml file of your robot's moveit_config
+  // or set explicit factors in your code if you need your robot to move faster.
+  move_group.setMaxVelocityScalingFactor(0.05);
+  move_group.setMaxAccelerationScalingFactor(0.05);
+
+  moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+  bool success = (move_group.plan(my_plan) == moveit::core::MoveItErrorCode::SUCCESS);
+  RCLCPP_INFO(LOGGER, "Visualizing plan 1 (joint space goal with cost function) %s", success ? "" : "FAILED");
+
+  // Visualize the plan in RViz:
+  visual_tools.deleteAllMarkers();
+  visual_tools.publishAxisLabeled(target_pose, "goal_pose");
+  visual_tools.publishText(text_pose, "Joint_Space_Goal", rvt::WHITE, rvt::XLARGE);
+  visual_tools.publishTrajectoryLine(my_plan.trajectory, joint_model_group);
+  visual_tools.trigger();
+
+  // Uncomment if you would like to execute the motion
+  /*
+  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to execute the motion");
+  move_group.execute(my_plan);
+  */
+
+  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to continue.");
+
+  // We can also specify cost functions when computing robot trajectories that must follow a cartesian path.
+  // First, let's change the target pose for the end of the path, so that if the previous motion plan was executed,
+  // we still have somewhere to move.
+  target_pose.position.y += 0.4;
+  target_pose.position.z -= 0.1;
+  target_pose.orientation.w = 0;
+  target_pose.orientation.x = -1.0;
+  Eigen::Isometry3d target;
+  tf2::fromMsg(target_pose, target);
+
+  auto start_state = move_group.getCurrentState(10.0);
+  std::vector<moveit::core::RobotStatePtr> traj;
+  moveit::core::MaxEEFStep max_eef_step(0.01, 0.1);
+  // Here, we're effectively disabling the jump threshold for joints. This is not recommended on real hardware.
+  moveit::core::JumpThreshold jump_thresh(0.0);
```

Comment:
@sea-bass Looks to me like jump checking is disabled completely (see [here](https://github.com/ros-planning/moveit2/blob/dd00f075ffc9b18b6311dc25102d3050cc9d6102/moveit_core/robot_state/src/cartesian_interpolator.cpp#L222)) but I don't know why :shrug: I guess the cost function prevent joint jumps here, since we're minimizing joint motions